src/system-health/health_checker/sysmonitor.py

```diff
@@ -0,0 +1,634 @@
+                continue
+            (key, op, cfvs) = cst.pop()
+            key_ext = key+".service"
+            #logger.log_info("featureevent received:{}".format(key_ext))
+            timestamp="{}".format(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+            msg={"unit": key_ext,"evt_src":"feature","time":timestamp}
+            self.task_notify(msg)
+            time.sleep(0.1)
+        
+        #logger.log_info("Stop STATE_DB monitoring loop")
+        
+    def task_worker(self):
+        while not self.task_stopping_event.is_set():
+            try:
+                self.task_worker_statedb()
+            except Exception as e:
+                logger.log_error("task_worker_statdb exited- {}, retry".format(str(e)))
+            time.sleep(1)
+    
+
+    def task_run(self):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_process = multiprocessing.Process(target=self.task_worker)
+        self.task_process.start()
+
+    def task_notify(self, msg):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_queue.put(msg)
+        time.sleep(0.001)
+
+    def task_stop(self):
+        self.task_stopping_event.set()
+        self.task_process.join()
+
+
+class MonitorSystemBusTask(object):
+    def __init__(self):
+        self.task_queue = mpmgr.Queue()
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def __init__(self,myQ):
+        self.task_queue = myQ
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def OnJobRemoved(self, id, job, unit, result):
+        #print('{}: Job Removed: {}, {}, {} '.format( id, job, unit, result))
+        if result == "done":
+            #logger.log_info("Received systemd service event for {}".format(unit))
+            timestamp="{}".format(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+            msg={"unit": unit,"evt_src":"sysbus","time":timestamp}
+            self.task_notify(msg)
+            return
+
+    #Sub process for listening the systemd event on dbus
+    def subscribe_sysbus(self):
+        import dbus
+        from gi.repository import GObject
+        from dbus.mainloop.glib import DBusGMainLoop
+
+        DBusGMainLoop(set_as_default=True)
+
+        bus = dbus.SystemBus()
+        systemd = bus.get_object('org.freedesktop.systemd1', '/org/freedesktop/systemd1')
+        manager = dbus.Interface(systemd, 'org.freedesktop.systemd1.Manager')
+
+        manager.Subscribe()
+        manager.connect_to_signal('JobRemoved', self.OnJobRemoved)
+
+        loop = GObject.MainLoop()
+        loop.run()
+
+    def task_worker(self):
+        #logger.log_info("Start Listening to systemd bus (pid {0})".format(os.getpid()))
+        update_proc_name('sysmonitor_sysbus')
+        
+        while not self.task_stopping_event.is_set():
+            self.subscribe_sysbus()
```

Comment:
I am not familiar with python dbus library, but are you sure that you want to create the bus object and subscribe it every 1 second? This is not event driven, this is polling.

Reply:
https://dbus.freedesktop.org/doc/dbus-python/tutorial.html#id4

(gdb) py-bt
Traceback (most recent call first):
  File "/usr/local/lib/python3.7/dist-packages/gi/overrides/GLib.py", line 497, in run
    super(MainLoop, self).run()
  File "/usr/local/bin/sysmonitor.py", line 806, in subscribe_sysbus
    loop.run()
  File "/usr/local/bin/sysmonitor.py", line 813, in task_worker
    self.subscribe_sysbus()
  File "/usr/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/usr/lib/python3.7/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.7/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.7/multiprocessing/context.py", line 277, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.7/multiprocessing/context.py", line 223, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/usr/lib/python3.7/multiprocessing/process.py", line 112, in start
    self._popen = self._Popen(self)
  File "/usr/local/bin/sysmonitor.py", line 820, in task_run
    self.task_process.start()
--Type <RET> for more, q to quit, c to continue without paging--
  File "/usr/local/bin/sysmonitor.py", line 1753, in system_service
    monitor_system_bus.task_run()
  File "/usr/local/bin/sysmonitor.py", line 1830, in <module>
    system_service()
(gdb)